src/main/java/de/dennisguse/opentracks/services/TrackDeleteService.java

```diff
@@ -0,0 +1,159 @@
+        sendResult(0, trackIds.size());
+        serviceExecutor.execute(() -> {
+            ContentProviderUtils contentProviderUtils = new ContentProviderUtils(this);
+            for (int i = 0; i < trackIds.size(); i++) {
+                contentProviderUtils.deleteTrack(this, trackIds.get(i));
+                updateNotification(i + 1, trackIds.size());
+                sendResult(i + 1, trackIds.size());
+            }
+            stopSelf();
+        });
+    }
+
+    private void sendResult(int deletes, int total) {
+        if (deleteResultObservable != null) {
+            deleteResultObservable.postValue(new DeleteStatus(deletes, total));
+        }
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return binder;
+    }
+
+    public LiveData<DeleteStatus> getDeletingStatusObservable() {
+        return deleteResultObservable;
+    }
+
+    /**
+     * Starts and shows the notification.
+     *
+     * @param tracksToDelete number of tracks to be deleted.
+     */
+    private void createAndShowNotification(int tracksToDelete) {
+        notificationManager = (NotificationManager) this.getSystemService(Context.NOTIFICATION_SERVICE);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel notificationChannel = new NotificationChannel(CHANNEL_ID, this.getString(R.string.app_name), NotificationManager.IMPORTANCE_LOW);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                notificationChannel.setAllowBubbles(true);
+            }
+
+            notificationManager.createNotificationChannel(notificationChannel);
+        }
+
+        notificationBuilder = new NotificationCompat.Builder(this, CHANNEL_ID);
+        notificationBuilder
+                .setContentTitle(this.getString(R.string.track_delete_progress_message))
+                .setContentText(this.getString(R.string.track_delete_progress, 0, tracksToDelete))
+                .setSmallIcon(R.drawable.ic_logo_color_24dp)
+                .setProgress(tracksToDelete, 0, false);
+
+        startForeground(NOTIFICATION_ID, notificationBuilder.build());
+    }
+
+    /**
+     * Updates notification progress.
+     *
+     * @param progress number of tracks already deleted.
+     * @param total    total of tracks to be deleted.
+     */
+    private void updateNotification(int progress, int total) {
+        notificationBuilder.setProgress(total, progress, false);
+        notificationBuilder.setContentText(this.getString(R.string.track_delete_progress, progress, total));
+        notificationManager.notify(NOTIFICATION_ID, notificationBuilder.build());
+    }
+
+    public class Binder extends android.os.Binder {
+
+        private Binder() {
+            super();
+        }
+
+        public TrackDeleteService getService() {
+            return TrackDeleteService.this;
+        }
+    }
+
+    public static class DeleteStatus {
+        private int progress;
+        private int max;
```

Comment:
max is count of tracks to be deleted and progress the current one?

Reply:
That's right.
I've used the same names that Android in `BuilderNotification.setProgress(int max, int progress, int indeterminate)`